tests/test_fairworkflow.py

```diff
@@ -332,3 +332,68 @@ def test_publish_as_nanopub_no_modifications(self, mock_publish, test_workflow):
         pubinfo = test_workflow.publish_as_nanopub()
         assert mock_publish.call_count == 0
         assert pubinfo['nanopub_uri'] is None
+
+    def test_workflow_construction_and_execution(self):
+        """
+        Construct a workflow using the is_fairstep and is_fairworkflow decorators
+        and check that execution and returned provenance is as expected.
+        """
+        @is_fairstep(label='Addition')
+        def add(a:float, b:float) -> float:
+            """Adding up numbers."""
+            return a + b
+
+        @is_fairstep(label='Subtraction')
+        def sub(a: float, b: float) -> float:
+            """Subtracting numbers."""
+            return a - b
+
+        @is_fairstep(label='Multiplication')
+        def mul(a: float, b: float) -> float:
+            """Multiplying numbers."""
+            return a * b
+
+        @is_fairstep(label='A strange step with little use')
+        def weird(a: float, b:float) -> float:
+            """A weird function"""
+            return a * 2 + b * 4
+
+        @is_fairworkflow(label='My Workflow')
+        def my_workflow(in1, in2, in3):
+            """
+            A simple addition, subtraction, multiplication workflow
+            """
+            t1 = add(in1, in2)
+            t2 = sub(in1, in2)
+            t3 = mul(weird(t1, in3), t2)
+            return t3
+
+        fw = my_workflow(1, 4, 3)
+        assert isinstance(fw, FairWorkflow)
+
+        # TODO: Find out why execution hangs only when running in pytest. May be to do with the threading.
+        #result, prov = fw.execute(num_threads=1)
+        #assert result == -66
+        #assert isinstance(prov, Publication)
+
+    def test_workflow_serialization(self):
```

Comment:
I guess you test serialization of a **complex** object here, maybe make that explicit.

Reply:
Agreed. Changed the name.